### PR TITLE
Add Integration tests and address n+1 problem

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following changes.

- Solved N + 1 problem when fetching all posts and their comments for a user.
Log for User Post index page before adding eager loading 
posts index before

Log for User Post index page after adding eager loading 
posts index after

Log for Post show page before adding eager loading 
post and comments before

Log for Post show page after adding eager loading 
posts show after

- Used Capybara to write integration tests for each view. 
`User index page`
`user show page`
`User post index page`
`Post show page`

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      expect(page).to have_content('Abdul')
+    end
+
+    it 'displays the number of posts the user has written' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('2')
+    end
+
+    it 'displays the post title' do
+      visit user_posts_path(@user)
```
Comment (youmari) [inline spec/views/post_index_spec.rb:33]: - [ ] Your tests are failing please make sure to fix them so they are green in order to complete the requirements 
Screenshot 2023-02-01 at 16 14 09

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @JonahKayizzi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/post_index_spec.rb (hunk 1 above)
Comment (JonahKayizzi) [reply spec/views/post_index_spec.rb:33]: Hello @youmari , 
Thanks for the review, however, the tests are running well on my end
As you have advised, I have attached a screenshot for the next reviewer

image

**Review by omarsalem7 — APPROVED** [review]
Hi @JonahKayizzi 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**